packages/trader/src/App/Components/Elements/VideoStream/video-stream.tsx

```diff
@@ -1,75 +1,299 @@
-    preload?: string;
-    primary_color?: string;
+type TVideoPlayerProps = {
     src: string;
-    start_time?: string | number;
-    test_id?: string;
+    is_mobile?: boolean;
+    data_testid?: string;
 };
 
-/* TODO [maryia-deriv]: This temporary component was created because the official Stream component
-   doesn't support disablePictureInPicture. Please remove it when creating custom player
-   because disablePictureInPicture will not matter when we disable controls and add our own ones instead. */
-const VideoStream = ({
-    ad_url,
-    allow_full_screen = true,
-    autoplay,
-    controls,
-    default_text_track,
-    disable_picture_in_picture,
-    letterbox_color = 'transparent', // unsets the default black background of the iframe
-    loop,
-    muted,
-    poster,
-    preload = 'auto',
-    primary_color,
-    src,
-    start_time,
-    test_id,
-    width = '100%',
-    height = '100%',
-    ...props
-}: TVideoStreamProps) => {
-    const params = [
-        poster && `poster=${encodeURIComponent(poster)}`,
-        ad_url && `ad-url=${encodeURIComponent(ad_url)}`,
-        default_text_track && `defaultTextTrack=${encodeURIComponent(default_text_track)}`,
-        primary_color && `primaryColor=${encodeURIComponent(primary_color)}`,
-        letterbox_color && `letterboxColor=${encodeURIComponent(letterbox_color)}`,
-        start_time && `startTime=${start_time}`,
-        muted && 'muted=true',
-        preload && `preload=${preload}`,
-        loop && 'loop=true',
-        autoplay && 'autoplay=true',
-        !controls && 'controls=false',
-    ]
-        .filter(Boolean)
-        .join('&');
+const VideoPlayer = ({ src, is_mobile, data_testid }: TVideoPlayerProps) => {
+    const should_autoplay =
+        checkUserBrowser() !== 'Safari' || (is_mobile && mobileOSDetect() !== 'iOS' && mobileOSDetect() !== 'unknown');
+
+    const [is_playing, setIsPlaying] = React.useState(false);
+    const [is_ended, setIsEnded] = React.useState(false);
+    const [current_time, setCurrentTime] = React.useState<number>();
+    const [video_duration, setVideoDuration] = React.useState<number>();
+    const [shift_X, setShiftX] = React.useState<number>(0);
+    const [show_controls, setShowControls] = React.useState(!should_autoplay);
+
+    const video_ref = React.useRef<StreamPlayerApi>();
+    const progress_bar_ref = React.useRef<HTMLDivElement>(null);
+    const animation_ref = React.useRef<number>(0);
+    const timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const is_dragging = React.useRef<boolean>(false);
+
+    const onVolumeChange = (new_value: number) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.volume = new_value;
+    };
+
+    const toggleMute = (new_value: boolean) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.muted = new_value;
+    };
+
+    const onPlaybackRateChange = (new_value: number) => {
+        if (!video_ref?.current) return;
```

Comment:
```suggestion
        if (!video_ref.current) return;
```

Reply:
Thank you, already fixed according to Bala suggestion (he left the same comment)